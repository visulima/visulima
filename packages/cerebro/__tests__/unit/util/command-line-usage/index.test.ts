import { describe, expect, it } from "vitest";

import commandLineUsage from "../../../../src/util/command-line-usage";

describe("line-usage", () => {
    it("should render a typical output", () => {
        expect.assertions(1);

        const definitions = [
            {
                alias: "h",
                description: "Display this usage guide.",
                group: "one",
                name: "help",
                type: Boolean,
            },
            {
                defaultOption: true,
                description: "The input files to process",
                group: "one",
                multiple: true,
                name: "src",
                type: String,
            },
            {
                alias: "t",
                description: "Timeout value in ms",
                name: "timeout",
                type: Number,
            },
        ];

        const sections = [
            {
                content: "Generates something very important.",
                header: "a typical app",
            },
            {
                header: "Option list",
                optionList: definitions,
            },
        ];

        expect(commandLineUsage(sections)).toMatchSnapshot();
    });

    it("should render a empty out if empty sections are provided", () => {
        expect.assertions(1);

        expect(commandLineUsage([])).toBe("");
    });
});
