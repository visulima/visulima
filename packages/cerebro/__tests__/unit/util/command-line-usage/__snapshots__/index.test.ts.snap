// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`line-usage > should render a typical output 1`] = `
"
[1ma typical app[22m

   Generates something very important.                                           

[1mOption list[22m

[90m [39m  [1m-h[22m, [1m[33m--help[39m[22m [4mboolean[24m                    [90m [39m  Display this usage guide.             
[90m [39m  [1m[33m--src[39m[22m [4mstring[][24m                        [90m [39m  The input files to process            
[90m [39m  [1m-t[22m, [1m[33m--timeout[39m[22m [4mnumber[24m                  [90m [39m  Timeout value in ms                   
"
`;
